backend-comparison/src/burnbenchapp/auth.rs

```diff
@@ -0,0 +1,178 @@
+use reqwest;
+use std::io::Write;
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{self, File},
+    path::{Path, PathBuf},
+};
+
+pub(crate) static CLIENT_ID: &str = "Iv1.84002254a02791f3";
+static GITHUB_API_VERSION_HEADER: &str = "X-GitHub-Api-Version";
+static GITHUB_API_VERSION: &str = "2022-11-28";
+
+/// Return the file path for the auth cache on disk
+pub(crate) fn get_auth_cache_file_path() -> PathBuf {
+    let home_dir = dirs::home_dir().expect("an home directory should exist");
+    let path_dir = home_dir.join(".cache").join("burn").join("burnbench");
+    #[cfg(test)]
+    let path_dir = path_dir.join("test");
+    let path = Path::new(&path_dir);
+    path.join("token.txt")
+}
+
+/// Returns true if the token is still valid
+pub(crate) fn verify_token(token: &str) -> bool {
+    let client = reqwest::blocking::Client::new();
+    let response = client
+        .get("https://api.github.com/user")
+        .header(reqwest::header::USER_AGENT, "burnbench")
+        .header(reqwest::header::ACCEPT, "application/vnd.github+json")
+        .header(reqwest::header::AUTHORIZATION, format!("Bearer {}", token))
+        .header(GITHUB_API_VERSION_HEADER, GITHUB_API_VERSION)
+        .send();
+    match response {
+        Ok(resp) => resp.status().is_success(),
+        Err(_) => false,
+    }
```

Comment:
We can simplify this with `map_or` function

Reply:
Got it, leveling up my Rust fu ! 😁